core/src/types/list.rs

```diff
@@ -81,42 +90,73 @@ impl Stream for Lister {
         }
 
         if let Some(fut) = self.stating.as_mut() {
-            let (path, rp) = ready!(fut.poll_unpin(cx));
+            let entry = ready!(fut.poll_unpin(cx));
 
             // Make sure we will not poll this future again.
             self.stating = None;
-            let metadata = match rp {
-                Ok(rp) => rp.into_metadata(),
-                Err(err) => {
-                    self.errored = true;
-                    return Poll::Ready(Some(Err(err)));
-                }
-            };
 
-            return Poll::Ready(Some(Ok(Entry::new(path, metadata))));
+            return Poll::Ready(Some(entry));
```

Comment:
If returning entry is an error, we should set `self.errored = true;` accordingly.

Reply:
Drop this block as no use.